packages/venia-ui/lib/components/SavedPaymentsPage/creditCard.js

```diff
@@ -0,0 +1,81 @@
+import React, { useCallback, useMemo } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { Trash2 as DeleteIcon } from 'react-feather';
+
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+
+import Icon from '../Icon';
+
+import defaultClasses from './creditCard.css';
+import LinkButton from '../LinkButton';
+
+/**
+ * Enumerated list of supported credit card types from
+ *
+ * https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Payment/view/base/web/js/model/credit-card-validation/credit-card-number-validator/credit-card-type.js
+ */
+const cardTypeMapper = {
+    AE: 'American Express',
+    AU: 'Aura',
+    DI: 'Discover',
+    DN: 'Diners',
+    ELO: 'Elo',
+    HC: 'Hipercard',
+    JCB: 'JCB',
+    MC: 'MasterCard',
+    MD: 'Maestro Domestic',
+    MI: 'Maestro International',
+    UN: 'UnionPay',
+    VI: 'Visa'
+};
+
+const CreditCard = props => {
+    const { classes: propClasses, details } = props;
+    const classes = mergeClasses(defaultClasses, propClasses);
+
+    const number = `**** ${details.maskedCC} \u00A0\u00A0 ${cardTypeMapper[
+        details.type
+    ] || ''}`;
+    const cardExpiryDate = useMemo(() => {
+        const [month, year] = details.expirationDate.split('/');
+        const shortMonth = new Date(+year, +month - 1).toLocaleString(
+            'default',
+            { month: 'short' }
+        );
+
+        return `${shortMonth}. ${year}`;
+    }, [details.expirationDate]);
+
+    // Should be moved to a talon in the future
+    const handleDelete = useCallback(() => {}, []);
+    const deleteButton = (
+        <LinkButton
+            classes={{ root: classes.deleteButton }}
+            onClick={handleDelete}
+        >
+            <Icon classes={{ icon: undefined }} size={16} src={DeleteIcon} />
+            <span className={classes.deleteText}>
+                <FormattedMessage
+                    id={'storedPayments.delete'}
+                    defaultMessage={'Delete'}
+                />
+            </span>
+        </LinkButton>
+    );
```

Comment:
If the delete functionality is not included as part of this functionality I would not include the button in the UI.

Reply:
We do have the mutation to delete a payment method, `DeletePaymentTokenOutput`. This functionality will be handled in https://jira.corp.magento.com/browse/PWA-638